### DISCUSSION TITLE
High CPU Utilization when Frigate running on HA as Addon

### DISCUSSION THREAD

**Discussion by MKSEagle75** [blakeblackshear/frigate#8217] [General]
I am trying to get frigate running properly.  I am able to see my camera (just 1 currently) and create masks etc but my CPU is running very high...if I turn off frigate it returns to normal.  I am running frigate as an Addon to TrueNas Scale in my Home Assistant instance.  Any assistance would be appreciated.

**Comment by NickM-27** [comment]
you should start by showing your config, https://github.com/blakeblackshear/frigate/issues/3860 is also a good read

**Reply by MKSEagle75** [reply]
Appologies...see attached.
![Screen Shot 2023-10-18 at 9 35 09 AM](https://github.com/blakeblackshear/frigate/assets/61063837/e0568221-b3fa-4c50-bac2-38bc397975e0)

**Reply by NickM-27** [reply]
you should be running detect on a sub stream not the main stream

**Reply by MKSEagle75** [reply]
Ok I changed that see attached...still same...I am running on server grade hardware I will also add a screenshot of that
![Screen Shot 2023-10-18 at 10 01 25 AM](https://github.com/blakeblackshear/frigate/assets/61063837/52ceb141-8082-4ab5-b878-abb251647f64)

![Screen Shot 2023-10-18 at 10 00 56 AM](https://github.com/blakeblackshear/frigate/assets/61063837/cb4aa132-260b-4d91-b29c-8715c986323c)

**Reply by NickM-27** [reply]
> I am running on server grade hardware I will also add a screenshot of that

then your hwaccel args are incorrect. you also need to change your masks because they are now incorrect as well due to the change in detect resolution

**Comment by MKSEagle75** [comment]
ok what hwaccel args should I use?  I changed my masks...also it seems to want to keep recording my car in my driveway constantly

**Reply by NickM-27** [reply]
both of these things are covered in the docs

https://docs.frigate.video/configuration/hardware_acceleration
https://docs.frigate.video/configuration/record

**Comment by MKSEagle75** [comment]
I put the intel hwaccel args in and I get this error
2023-10-18 14:21:33.033660202  [2023-10-18 10:21:33] frigate.video                  ERROR   : driveway: Unable to read frames from ffmpeg process.

**Reply by NickM-27** [reply]
you'll need to let it run longer so full logs can be read, most likely the docker or VM config is not mapping the GPU correctly

**Reply by MKSEagle75** [reply]
That recordings part of the documentation didn't say anything about not recording stationary objects like my car in the driveway

**Reply by NickM-27** [reply]
It specifically mentions that scenario in this section https://docs.frigate.video/configuration/record#what-do-the-different-retain-modes-mean

In this case why do you think your car is being recorded? The event being in progress on the events page does not mean it is being constantly recorded.

**Reply by MKSEagle75** [reply]
because there are multiples of just my car in events

**Reply by NickM-27** [reply]
A screenshot may be helpful to see to understand what is happening exactly

**Reply by MKSEagle75** [reply]
agreed here you go.
![Screen Shot 2023-10-18 at 10 44 23 AM](https://github.com/blakeblackshear/frigate/assets/61063837/b415ddd2-9c3a-4b0d-8652-58021ff5c63a)

**Reply by MKSEagle75** [reply]
I deleted the other ones already

**Reply by NickM-27** [reply]
right, this is a known behavior in frigate 0.12 because the object tracking was very simple. Sometimes it would get mixed up on which car is which and it would think the car that is driving by is the car that was in the driveway. 

The object tracker as well as other areas of this logic have been rewritten in 0.13 (currently in beta)

**Reply by MKSEagle75** [reply]
and there is a mask on the upper corner but it still picked up a car driving past
![Screen Shot 2023-10-18 at 10 46 18 AM](https://github.com/blakeblackshear/frigate/assets/61063837/2231ddf4-d034-4f47-a549-5c724a777c66)

**Reply by NickM-27** [reply]
to be clear `( in progress)` does not mean it is being continuously recorded, recordings are only retained according to the recording config

**Reply by NickM-27** [reply]
a mask will not stop objects from being detected and using masks like this is highly discouraged in the docs. That is what [required zones are for](https://docs.frigate.video/configuration/zones#restricting-events-to-specific-zones)
